src/session.rs

```diff
@@ -262,6 +275,8 @@ impl BackgroundSession {
         })
     }
     /// Unmount the filesystem and join the background thread.
+    /// # Panics
+    /// Panics if the background thread can't be recovered.
```

Comment:
Let's remove this. It's an implementation detail

Reply:
I think we should only document things that are part of the public API, rather than implementation details. That is they're not going to change

Reply:
I removed most of them, but I didn't remove this one. Joining a background thread can panic, and sometimes should, panic. For example,

- A panic in the session thread would propagate to the main thread on `join()`.
- A deadlocked session thread may or may not cause a panic on `join()`, depending on the OS.

These are both just a normal part of joining a background thread. The underlying call to `thread::join()` even has its own `# Panic` documentation. It would be disingenuous to imply that our function is somehow free from panics.

I did improve the `# Panic` section by clarifying an example.